workspaces/arborist/lib/override-set.js

```diff
@@ -55,7 +93,9 @@ class OverrideSet {
         return rule
       }
 
-      let spec = npa(`${edge.name}@${edge.spec}`)
+      // We need to use the rawSpec here, because the spec has the overrides applied to it already.
+      // rawSpec can be undefined, so we need to use the fallback value of spec if it is.
+      let spec = npa(`${edge.name}@${edge.rawSpec || edge.spec}`)
```

Comment:
I'm making an assumption here that we can fallback

Reply:
@owlstronaut

Just below this
```js
 if (spec.type === 'git') {
        if (spec.gitRange && semver.intersects(spec.gitRange, rule.keySpec)) 
```
What condition guarantees `rule.keySpec` is set. It seems if there's no `parent` it's not set. Is that assumed here? If so can you provide context/details.

Reply:
@owlstronaut NM I see it. They come from rulesets which resolve from children of others (so they have parents).